### ISSUE TITLE
`For` directive breaks after code edits in pub serve

### ISSUE THREAD

**Issue by jakemac53** [angular/angular#1617]
When using the `For` directive in a component it works when you first start up pub serve, but once you make any code changes you start getting the following error:

```Exception: Uncaught Error: Cannot find setter of```

I am using it like:

```html
<core-item *for="#message of messages">
  <div>{{message}}</div>
</core-item>
```

**Comment by kegluneq** [comment]
Very, very strange behavior. I'm going to postpone really looking into this until we have the full render compiler pipeline running in the transformer.

In the meantime, you can manually populate the `reflector` as a workaround:

```
reflector.registerSetters({'of': (o, v) => o.of = v});
```

**Comment by kegluneq** [comment]
TL;DR: Swapped the effort labels because this has something to do with the internals of the Angular `Compiler`. I would guess it will take on the order of days to track down exactly what's happening here, but it may go faster if someone with knowledge of the `Compiler` is willing to dig in.

### Details

What this means is that the Angular `Compiler` is requesting different things when we run it at transform time than at runtime. During transformation, we "dry run" the `Compiler` and note what getters, setters, and methods it requests, then we generate code for those.

In this case, the `Compiler` does not request a setter for `of` at transform time but does request one at runtime. I'm not sure why

1. It would think `of` could be set given that template
2. It would think that at runtime only, not during transformation

**Comment by kegluneq** [comment]
This will almost certainly be fixed by https://github.com/angular/angular/issues/2359

**Comment by kegluneq** [comment]
I'm going to close this as fixed by d1b35f9 and 87b3b71, but let me know if it shows up again.
